riotfile.py

```diff
@@ -79,6 +79,82 @@ def select_pys(min_version: str = MIN_PYTHON_VERSION, max_version: str = MAX_PYT
     return [version_to_str(version) for version in SUPPORTED_PYTHON_VERSIONS if min_version <= version <= max_version]
 
 
+def create_memory_allocator_venvs() -> List[Venv]:
+    """Create venv variants for testing different Python memory allocators."""
+    import ctypes
+
+    allocator_configs = [
+        {
+            "name_suffix": "pymalloc",
+            "env": {"PYTHONMALLOC": "pymalloc"},
+            "description": "Python's default small object allocator",
+        },
+        {"name_suffix": "malloc", "env": {"PYTHONMALLOC": "malloc"}, "description": "System malloc allocator"},
+        {
+            "name_suffix": "malloc-debug",
+            "env": {"PYTHONMALLOC": "malloc_debug"},
+            "description": "System malloc with debug hooks",
+        },
+        {
+            "name_suffix": "pymalloc-debug",
+            "env": {"PYTHONMALLOC": "pymalloc_debug"},
+            "description": "Python malloc with debug hooks",
+        },
+    ]
+
+    # Conditionally add jemalloc variant if the library is available
+    jemalloc_names = [
+        "libjemalloc.so.2",  # Ubuntu/Debian
+        "libjemalloc.so.1",  # Older versions
+        "libjemalloc.so",  # Generic
+        "libjemalloc.dylib",  # macOS (though less common)
+    ]
+
+    for lib_name in jemalloc_names:
+        try:
+            # Try to load jemalloc library (OSError is raised if not found)
+            ctypes.CDLL(lib_name)
+
+            allocator_configs.append(
+                {
+                    "name_suffix": "jemalloc",
+                    "env": {"PYTHONMALLOC": "malloc", "LD_PRELOAD": lib_name},
+                    "description": "jemalloc high-performance allocator",
+                }
+            )
+
+            break
+        except (OSError, AttributeError):
+            # Library not found or ctypes issue - try next name
+            continue
+
+    return [
```

Comment:
Would it be possible to just specify env var for different allocators and run that as part of the existing venvs? and only for memory allocators?

This would result in running all profiling tests with different allocators.

Reply:
Good idea, let me check.